packages/salesforcedx-vscode-core/src/commands/forceOrgOpen.ts

```diff
@@ -6,17 +6,87 @@
  */
 
 import {
+  CliCommandExecutor,
   Command,
   SfdxCommandBuilder
 } from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import { ContinueResponse } from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
 import { nls } from '../messages';
+import { telemetryService } from '../telemetry';
+import { getRootWorkspacePath, isSFDXContainerMode } from '../util';
 import {
   EmptyParametersGatherer,
   SfdxCommandlet,
   SfdxCommandletExecutor,
   SfdxWorkspaceChecker
 } from './commands';
 
+type CLIOrgData = {
+  status: number;
+  result: {
+    orgId: string;
+    url: string;
+    username: string;
+  };
+};
+
+class ForceOrgOpenContainerExecutor extends SfdxCommandletExecutor<{}> {
+  public build(data: {}): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_org_open_default_scratch_org_text'))
+      .withArg('force:org:open')
+      .withLogName('force_org_open_default_scratch_org')
+      .withArg('--urlonly')
+      .withJson()
+      .build();
+  }
+
+  private buildUserMessageWith(orgData: CLIOrgData): string {
+    return nls.localize(
+      'force_org_open_container_mode_message_text',
+      orgData.result.orgId,
+      orgData.result.username,
+      orgData.result.url
+    );
+  }
+
+  private openBrowserWithCliResponse(cliResponseJSON: string | Buffer) {
+    try {
+      const cliOrgData: CLIOrgData = JSON.parse(cliResponseJSON.toString());
+      const authenticatedOrgUrl: string = cliOrgData.result.url;
+
+      channelService.appendLine(this.buildUserMessageWith(cliOrgData));
+
+      if (authenticatedOrgUrl) {
+        vscode.env.openExternal(vscode.Uri.parse(authenticatedOrgUrl));
+      }
+    } catch (e) {
+      telemetryService.sendErrorEvent(e.message, e.stack);
```

Comment:
I think it would be worthwhile also presenting so error to the user if this fails to open. That way the user knows that something has failed and can report to use the error message.

Maybe something like

```
channelService.appendLine(`Failed to open org: ${e.message}`);
```

Reply:
We should avoid adding `e.stack` to the metrics since it might contain user data like file paths, etc.

Reply:
Good catch. @jonnyhork this is another place where you are breaking the contract by doing something different other than what we have already at https://github.com/forcedotcom/salesforcedx-vscode/blob/develop/packages/salesforcedx-vscode-core/src/commands/baseDeployCommand.ts#L81

Please double-check that you are not doing anything that deviates from that. And If you are call that out explicitly, such things could be missed during code reviews since we are not always going to compare each line of your code with the base method (it doesn’t show up in the diff).